Code/Framework/AzToolsFramework/AzToolsFramework/FocusMode/FocusModeSystemComponent.cpp

```diff
@@ -112,4 +124,54 @@ namespace AzToolsFramework
         return AzToolsFramework::IsInFocusSubTree(entityId, m_focusRoot);
     }
 
+    void FocusModeSystemComponent::OnEntityInfoUpdatedAddChildEnd(AZ::EntityId parentId, AZ::EntityId childId)
+    {
+        // If the parent's entityId is in the list, add the child.
+        if (auto iter = AZStd::find(m_focusedEntityIdList.begin(), m_focusedEntityIdList.end(), parentId);
+            iter != m_focusedEntityIdList.end())
+        {
+            m_focusedEntityIdList.push_back(childId);
+        }
+    }
+
+    void FocusModeSystemComponent::OnEntityInfoUpdatedRemoveChildEnd([[maybe_unused]] AZ::EntityId parentId, AZ::EntityId childId)
+    {
+        // If the removed entityId is in the list, remove it.
+        if (auto iter = AZStd::find(m_focusedEntityIdList.begin(), m_focusedEntityIdList.end(), childId);
+            iter != m_focusedEntityIdList.end())
+        {
+            m_focusedEntityIdList.erase(iter);
+        }
+    }
+
+    void FocusModeSystemComponent::OnPrefabInstancePropagationEnd()
+    {
+        // Can't rely on any of the entities in the list to still exist, refresh the whole thing.
+        RefreshFocusedEntityIdList();
+    }
+
+    void FocusModeSystemComponent::RefreshFocusedEntityIdList()
```

Comment:
suggestion: For a future PR/change this would be a great candidate to profile and benchmark - if we had a friendlier way of iterating entity hierarchies we wouldn't need to rewrite this logic here, it would be great if this was cached or just much quick to traverse. Could be a great area to optimize in future (and it's awesome we have the tests now to make this exercise less perilous 😛)